Hotfix/spectate teleport(modular admin fixes)

xal3xhx opened the pull request:
modular admin fixes

IBlackI committed:
Changed modular_admin_spectate

Sergeant-Steve committed:
Fixed Spectate Teleport

Fixed Modular Admin Spectate Non-Colliding position bad number of arguments (hopefully)

Sergeant-Steve committed:
Actually fixed spectate teleport

Turns out I was wrong initially, who knew?

xal3xhx committed:
Update modular_admin_ghosts.lua

Sergeant-Steve committed:
Fixed some bugs

Fixed teleport and following bugs for spectate and player list, also added check to make sure that parent is nil on modular_admin_tag (I hope).

Commit:
Merge pull request #49 from xal3xhx/master

fixed crash when robot deconstructs a entity with a nil name

IBlackI commented on lualib/modular_admin/modular_admin_tag.lua:
```diff
@@ -72,7 +72,7 @@ function modular_admin_tag_gui_clicked(event)
 	if e ~= nil then
 		if p.admin then
-			if e.parent.name == modular_admin_get_menu(p).name then
+			if e.parent ~= nil and (e.parent.name == modular_admin_get_menu(p).name) then
```
e.parent ~= nil can be earlier at the e.valid check.

Sergeant-Steve replied:
I'm assuming that this is the right code to begin with, and given my lack of knowledge I assumed it would make sense to have it next to the parent part. If it should be moved higher then be my guest.

xal3xhx commented on lualib/modular_admin/modular_admin_spectate.lua:
```diff
@@ -156,7 +156,7 @@ function modular_admin_spectate_set_normal_teleport(p)
 	if global.modular_admin_spectate.player_spectator_state[p.index] == false then
 		p.print("Teleporting you to the location you are currently looking at.")
-		p.teleport(p.surface.find_non_colliding_position("player", pos))
+		p.teleport(p.surface.find_non_colliding_position("player", pos, 0, 1))
```
caused crash.. 
`190267.216 Error MainLoop.cpp:1035: Exception at tick 7693249: Error while running event level::on_tick (ID 0)
...factorio/server2/temp/currently-playing/lualib/event.lua:294: ...-playing/lualib/modular_admin/modular_admin_spectate.lua:111: bad argument #0 to 'find_non_colliding_position' (Wrong number of arguments.)`


`190267.216 Error ServerMultiplayerManager.cpp:96: MultiplayerManager failed: "Error while running event level::on_tick (ID 0)
...factorio/server2/temp/currently-playing/lualib/event.lua:294: ...-playing/lualib/modular_admin/modular_admin_spectate.lua:111: bad argument #0 to 'find_non_colliding_position' (Wrong number of arguments.)"`

Sergeant-Steve replied:
Fixed with latest commit. Also you highlighted the wrong faulting line. Line 111 had the bad argument in the stack trace, you highlighted line 159.

Sergeant-Steve committed:
Fixed crash for teleport

Removed redundant argument from find_non_colliding_position left behind from changing code previously.

Sergeant-Steve committed:
Merge remote-tracking branch 'origin/hotfix/spectate-teleport' into hotfix/spectate-teleport
